keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,731 @@
+    // DNS sub-domains separated by slashes.
+    //
+    // +required
+    Pool string `json:"pool" protobuf:"bytes,2,rep,name=pool"`
+
+    // Device references one device instance via its name in the driver's
+    // resource pool. It must be a DNS label.
+    //
+    // +required
+    Device string `json:"device" protobuf:"bytes,3,rep,name=device"`
+
+    // Conditions contains the latest observation of the device's state.
+    // If the device has been configured according to the class and claim
+    // config references, the `Ready` condition should be True.
+    //
+    // +optional
+    // +listType=atomic
+    Conditions []metav1.Condition `json:"conditions" protobuf:"bytes,4,opt,name=conditions"`
+
+    // Data contains arbitrary driver-specific data.
+    //
+    // +optional
+    Data runtime.RawExtension `json:"data,omitempty" protobuf:"bytes,5,opt,name=data"`
+
+    // NetworkData contains network-related information specific to the device.
+    //
+    // +optional
+    NetworkData NetworkDeviceData `json:"networkData,omitempty" protobuf:"bytes,6,opt,name=networkData"`
+}
+
+// NetworkDeviceData provides network-related details for the allocated device.
+// This information may be filled by drivers or other components to configure
+// or identify the device within a network context.
+type NetworkDeviceData struct {
+    // InterfaceName specifies the name of the network interface associated with
+    // the allocated device. This might be the name of a physical or virtual
+    // network interface.
+    //
+    // +optional
+    InterfaceName string `json:"interfaceName,omitempty" protobuf:"bytes,1,opt,name=interfaceName"`
+
+    // Addresses lists the network addresses assigned to the device's network interface.
+    // This can include both IPv4 and IPv6 addresses.
+    // The addresses are in the CIDR notation, which includes both the address and the
+    // associated subnet mask.
+    // e.g.: "192.0.2.0/24" for IPv4 and "2001:db8::/64" for IPv6.
+    //
+    // +optional
+    // +listType=atomic
+    Addresses []string `json:"addresses,omitempty" protobuf:"bytes,2,opt,name=addresses"`
+
+    // HWAddress represents the hardware address (e.g. MAC Address) of the device's network interface.
+    //
+    // +optional
+    HWAddress string `json:"hwAddress,omitempty" protobuf:"bytes,3,opt,name=hwAddress"`
+}
+```
+
+### User Stories (Optional)
+
+#### Story 1 - Network Device Status for Network Services
+
+As a Cloud Native Network Function (CNF) vendor, my network services must be
+integrated with the network devices configured in Pods. The configuration
+properties of these network devices are therefore essential to configure the
+network services. For example, the network services must be able to route
+traffic to pods over networks attached via the network devices, the IP
+addresses of the network device(s) must then be reflected in the
+`ResourceClaim.Status` allowing the network service controller(s) to access
+them.
```

Comment:
So this motivates having the IPs be available, but not the hardware addrs (or anything else). Is that correct?